streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -151,25 +151,68 @@ public String metricsScope() {
      * @param windowSize            size of the windows (cannot be negative)
      * @param retainDuplicates      whether or not to retain duplicates.
      * @return an instance of {@link WindowBytesStoreSupplier}
+     * @deprecated since 2.1 Use {@link Stores#persistentWindowStore(String, long, long, boolean, long)} instead
      */
+    @Deprecated
     public static WindowBytesStoreSupplier persistentWindowStore(final String name,
                                                                  final long retentionPeriod,
                                                                  final int numSegments,
                                                                  final long windowSize,
                                                                  final boolean retainDuplicates) {
+        if (numSegments < 2) {
+            throw new IllegalArgumentException("numSegments cannot must smaller than 2");
+        }
+
+        return persistentWindowStore(
+            name,
+            retentionPeriod,
+            windowSize,
+            retainDuplicates,
+            Math.max(retentionPeriod / (numSegments - 1), 60_000)
+        );
+    }
+
+    /**
+     * Create a persistent {@link WindowBytesStoreSupplier}.
+     * @param name                  name of the store (cannot be {@code null})
+     * @param retentionPeriod       length of time to retain data in the store (cannot be negative)
+     * @param windowSize            size of the windows (cannot be negative)
+     * @param retainDuplicates      whether or not to retain duplicates.
+     * @return an instance of {@link WindowBytesStoreSupplier}
+     */
+    public static WindowBytesStoreSupplier persistentWindowStore(final String name,
+                                                                 final long retentionPeriod,
+                                                                 final long windowSize,
+                                                                 final boolean retainDuplicates) {
+        return persistentWindowStore(name, retentionPeriod, windowSize, retainDuplicates, Math.max(retentionPeriod / 2, 60_000L));
```

Comment:
Could we use a const for `60_000L` than putting the magic number scattered in multiple places?

Reply:
Yes, we certainly can. Similar to my comment on the maintain-ms default, I think it's valuable for each decoupled component to have an independent constant. But in this case, I think we would actually benefit from having the constant, as it would have more than one use.

Reply:
@guozhangwang While thinking about this, I had a more fundamental question... Why is 1 minute the minimum segment interval?

Through some archaeology, I've determined that the old segment file naming scheme encoded the segment as a date, with one minute granularity. So I think that the restriction was originally put into place for this reason.

Nowadays, though, I don't think there is any such restriction. Perhaps instead of encoding the magic value, we can just banish it instead, by lifting this restriction?

Reply:
Honestly I cannot remember the exact reasons why we use 1minute as the magic number...

Just using retentionPeriod / num.segments (enforcing num.segments > 2) should be fine.

Reply:
If it's ok with you, I'd like to tackle this issue in a follow-on PR.

I've got the code to loosen the restriction and edited the tests to prove that everything still works with other segmentIntervals, but the diff is quite large and orthogonal to KIP-319.

Can we just leave it as a magic number for now?

Reply:
sounds good.